rust/lance/src/dataset/write/delete.rs

```diff
@@ -162,244 +273,324 @@ mod tests {
         data_storage_version: LanceFileVersion,
         #[values(false, true)] with_scalar_index: bool,
     ) {
-        fn sequence_data(range: Range<u32>) -> RecordBatch {
-            let schema = Arc::new(ArrowSchema::new(vec![
-                ArrowField::new("i", DataType::UInt32, false),
```

Comment:
The changes to this test and others seem unnecessary. Please leave all existing tests unchanged.